okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/connection/Exchange.kt

```diff
@@ -248,7 +248,10 @@ class Exchange(
     private val contentLength: Long,
   ) : ForwardingSink(delegate) {
     private var completed = false
+
+    // TODO bytesReceived or bytesWritten?
     private var bytesReceived = 0L
+    private var invokeStartEvent = true
```

Comment:
I think this class needs to know if it’s being used for the socket or a regular request body

Reply:
`isSocket` will become `true` on `upgradeToSocket()`, so `invokeStartEvent` will become `true`, too.